.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  unit_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 10.x
+
+      - name: Cache node modules
+        id: yarn-cache
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-hash-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install Packages
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
```

Comment:
Skipping `yarn install` is a bold move.

Unfortunately, to ensure all dependencies are valid, `yarn install` doesn't benefit from cache. Almost all the time it takes is verifying the integrity of all the files in `node_modules` and ensuring everything matches what's in the `yarn.lock` file.

Skipping makes the following assumptions:
- `yarn.lock` + OS are the only things that effect the contents of `node_modules`
- None of our CI scripts change the contents of `node_modules` _after_ `yarn install`
- Github Actions does file integrity check both when creating and when restoring cache files (failure may result in incomprehensible errors)
- None of our scripts use the contents of `node_modules` that is local to the module itself (`uuid` typically adds a `node_modules/.bin/uuid` command local to the module)

As long as these assumptions hold true, we save about 6 minutes on a build. In this script, that's 2/3 of the total CI time!

Absolute worst case: corrupt cache. Wasted time to find out it was a cache error. Destroy the cache.
Worst case: cache miss. Build takes an extra 1 minutes to create a cache.
Best case: cache hit. Build takes 6 minutes less.